backend/capellacollab/projects/toolmodels/modelsources/git/gitlab/handler.py

```diff
@@ -49,10 +50,23 @@ async def get_last_job_run_id_for_git_model(
                 pipeline_id,
                 job_name,
             ):
-                return handler.JobIdAttributes(project_id, job)
+                return job
 
         raise git_exceptions.GitPipelineJobNotFoundError(job_name=job_name)
 
+    def get_last_updated_for_file_path(
+        self, project_id: str, file_path: str, revision: str | None
+    ) -> datetime.datetime | None:
+        response = requests.get(
+            f"{self.git_instance.api_url}/projects/{project_id}/repository/commits?ref_name={revision or self.git_model.revision}&path={file_path}",
+            headers={"PRIVATE-TOKEN": self.git_model.password},
+            timeout=config["requests"]["timeout"],
+        )
+        response.raise_for_status()
+        if len(response.json()) == 0:
+            return None
```

Comment:
Here is another occurrence of `len(response.json()) == 0` where `git_exceptions.GitRepositoryFileNotFoundError` can be used.

Reply:
good catch